### DISCUSSION TITLE
topo_mask for Ensemble-Stat

### DISCUSSION THREAD

**Discussion by JiliDong-NOAA** [dtcenter/METplus#1638] [Configuration] [answered]
While working on verification for ensemble forecasts, I found Ensemble-Stat didn't take topo_mask dictionary, which works well for Point-Stat. EnsembleStat didn't crash but showed no impact of topo_mask. As we need to limit the ensemble verification to obs-model pairs with small terrain difference, I wonder if there is any way to use topo_mask for ensemble-stat or if my set up was incorrect? Here is the topo_mask section I added for MET config of EnsembleStat:

```
topo_mask = {
   flag               = TRUE;
   file_name          = [];
   field              = { name = "HGT"; level = "Z0"; }
   regrid             = { method = BILIN; width = 2; }
   use_obs_thresh     = ge-100&&le100;
   interp_fcst_thresh = ge-50&&le50;
}
```

Thanks,

Jili

**Comment by j-opatz** [comment]
Hello Jili,

And thank you for your question. Upon review of the [documentation](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html?highlight=topo_mask#topo-mask), I tend to agree with your assessment: topo_mask should work with ensemble-stat, as long as your observation dataset is point observations.

I'm going to tag @JohnHalleyGotway so he can provide a more succinct answer when he returns to the office, but in the meantime I do have two recommendations to provide.

The first is to double-check your `message_type_group_map` dictionary and ensure that the `SURFACE` key has the correct values attached to it. If your observation variable `HGT` does not come from one of the groups listed in the `val` entries (or if you forgot to include the `message_type_group_map` dictionary), then it will not be used for topography information.
Secondly, you may want to change out the level you're checking to `L0`. This is only to double check that `Z0` will not cause unforseen errors and to better match the documentation examples; I am skeptical that this will suddenly allow the logic to work, but it's worth a check.

**Comment by j-opatz** [comment, marked as answer]
Hi @JiliDong-NOAA, I wanted to update you on the situation:

John HG provided the correct information that ensemble-stat does **not** support the use of `topo_mask`.

However, he agrees that this is something that can be added to ensemble-stat and has created [an issue](https://github.com/dtcenter/MET/issues/1120) for the creation of this ability.

We are also looking to update the documentation so that details like these (i.e. tool-specific dictionaries and options) are easier to find and understand.

**Reply by JiliDong-NOAA** [reply]
Thanks for the info and help! I am looking forward to future updates on this. 

Jili
